### PR TITLE
bump: update go to v1.18.1

### PR DESCRIPTION
**Pull request by denis-tingaikin**
Signed-off-by: Denis Tingaikin <denis.tingajkin@xored.com>

## Motivation

SOFT update to go v1.18.1

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+
+require (
+	github.com/davecgh/go-spew v1.1.0 // indirect
+	github.com/golang/protobuf v1.4.0 // indirect
+	github.com/pmezard/go-difflib v1.0.0 // indirect
+	golang.org/x/net v0.0.0-20190311183353-d8887717615a // indirect
+	golang.org/x/sys v0.0.0-20190215142949-d0b11bdaac8a // indirect
+	golang.org/x/text v0.3.0 // indirect
+	google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55 // indirect
+	gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c // indirect
```
Comment (denis-tingaikin) [inline go.mod:20]: Note: go 1.17 the module graph had changed --> added  pruning/lazy loading.

See at  https://golang.org/doc/go1.17#go-command

File: .golangci.yml [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ---
 run:
   # concurrency: 6
+  go: "1.17"
```
Comment (denis-tingaikin) [inline .golangci.yml:4]: https://github.com/golangci/golangci-lint/issues/2649

Comment (edwarnicke) [inline .golangci.yml:4]: Why is this 1.17 instead of 1.18.1 ?

Comment (denis-tingaikin) [reply .golangci.yml:4]: golangci guys is not added support for 1.18. The work in progress see at golangci/golangci-lint#2649.
We'll update this as soon as golangci will be updated.

**Pull request merged**